src/App.jsx

```diff
@@ -14,45 +18,59 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button
-        type="button"
-        className="button is-info is-light"
-      >
-        Sort alphabetically
-      </button>
-
-      <button
-        type="button"
-        className="button is-success is-light"
-      >
-        Sort by length
-      </button>
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
+const prepareGoods = (goods, sortOrder, sortDirection) => {
+  const goodsCopy = [...goods];
+
+  switch (sortOrder) {
+    case ORDER.ALPHABETICALLY:
+      goodsCopy.sort((next, prev) => next.localeCompare(prev));
+      break;
+    case ORDER.BY_LENGTH:
+      goodsCopy.sort((next, prev) => next.length - prev.length);
+      break;
+    default:
+      break;
+  }
+
+  if (sortDirection === DIRECTION.OPPOSITE) {
+    goodsCopy.reverse();
+  }
+
+  return goodsCopy;
+};
+
+export const App = () => {
+  const [order, setOrder] = useState(ORDER.BY_DEFAULT);
+  const [direction, setDirection] = useState(DIRECTION.DIRECTLY);
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+  const switchDirection = () => {
+    if (direction === DIRECTION.DIRECTLY) {
+      setDirection(DIRECTION.OPPOSITE);
+    }
+
+    if (direction === DIRECTION.OPPOSITE) {
+      setDirection(DIRECTION.DIRECTLY);
+    }
+  };
+
+  const preparedGoods = prepareGoods(
+    goodsFromServer, order, direction,
+  );
+
+  return (
+    <div className="section content">
+      <Buttons
+        sortAlphabetically={() => setOrder(ORDER.ALPHABETICALLY)}
+        sortByLength={() => setOrder(ORDER.BY_LENGTH)}
+        reverse={() => switchDirection()}
+        reset={() => {
+          setOrder(ORDER.BY_DEFAULT);
+          setDirection(DIRECTION.DIRECTLY);
+        }}
```

Comment:
it is bad practise to make anonymous arrow functions in jsx.
make handler in component and pass reference here

Reply:
Is it OK use **reset={() => resetHandler()}** instead **reset={resetHandler}** ?